src/serializers/index.ts

```diff
@@ -45,19 +45,17 @@ const isObject = (value: unknown): boolean => {
   return value != null && (type === 'object' || type === 'function');
 };
 
-const createReqSerializer = (opts: SerializerOptions) => (request: Request) =>
-  isObject(request)
-    ? {
-        method: request.method,
-        url: request.url,
-        headers: omitProperties(
-          request.headers,
-          opts.omitHeaderNames ?? defaultOmitHeaderNames,
-        ),
-        remoteAddress: request?.socket?.remoteAddress,
-        remotePort: request?.socket?.remotePort,
-      }
-    : request;
+const createReqSerializer =
+  (serializeHeaders: SerializerFn) => (request: Request) =>
+    isObject(request)
+      ? {
+          method: request.method,
+          url: request.url,
+          headers: serializeHeaders(request.headers),
```

Comment:
Unifying the code paths here so we can benefit from the deduplication implemented in `createOmitPropertiesSerializer`